### ISSUE TITLE
Autocompletion issues with preprocessed templates

### ISSUE THREAD

**Issue by ltrzesniewski** [JetBrains/ForTea#28]
Here's something I noticed:

- Create any T4 preprocessed template, for instance:

```
<#@ template language="C#" #>
Hello!
```

- Run the "Preprocess Template" action

![image](https://user-images.githubusercontent.com/7913492/79602838-87121300-80eb-11ea-8fc1-3b9028558b60.png)

- Try to use it from C#:

```C#
var test = new Test();
test.TransformText();
```

- The IDE does not suggest `Test` in the autocompletion popup, but does suggest `TestBase`:

![image](https://user-images.githubusercontent.com/7913492/79603103-fdaf1080-80eb-11ea-9989-b41f7acd3c94.png)

- The `TransformText` method is not present in the suggestions either:

![image](https://user-images.githubusercontent.com/7913492/79603033-d9ebca80-80eb-11ea-8f66-65540bdf3340.png)

- Go to definition does not work on the type or the method.

Tested with Rider 2020.1, with the bundled 2020.1.157 version of the plugin.

**Comment by kirillgla** [comment]
Thank you for reporting that! Is that specific to preprocessed templates, or do you expect to see those symbols in completion in executable templates, too?

**Comment by ltrzesniewski** [comment]
I'm not sure I understand your question, as preprocessed templates _are_ executable templates in my understanding. I'm basically talking about the `TextTemplatingFilePreprocessor` generator, which always generates a class with a `TransformText` method.

**Comment by kirillgla** [comment]
Yes, I totally agree that those symbols should be visible in preprocessed files. By 'executable' I mean templates with `TextTemplatingFileGenerator` generator. That generator also generates a class and the `TransformText` method behind the scenes, but should they be visible in completion? I mean, in templates with `TextTemplatingFileGenerator` those symbols just look like implementation details that are not intended to be accessed from C# code in T4 file, aren't they?

**Comment by ltrzesniewski** [comment]
I don't expect to see the template class generated by `TextTemplatingFileGenerator` in the completion. This is an implementation detail as you say, and not part of the project. Only the *result* of template execution should be visible in the project.

Actually, the way I see it is pretty simple: executing a `TextTemplatingFileGenerator` or a `TextTemplatingFilePreprocessor` generator will produce an output file besides the `.tt` file. `TextTemplatingFilePreprocessor` always produces a `.cs` file, and `TextTemplatingFileGenerator` can produce any file type. I do *not* expect Rider to treat this file in *any* particular way. Just let the IDE handle it like any other file in the project. The template output file should not be of any concern to ForTea.

Currently, there is definitely some special treatment attached to the output of a T4 template. This issue along with the fixed #20 and #15 demonstrate that. If the output file was handled like any other file, none of these three issues would have occurred.

**Comment by kirillgla** [comment]
I agree. I'll fix that once I can; it shouldn't be difficult

**Comment by kirillgla** [comment]
Done. The fix should be publicly available in 2020.2 EAP 1

**Comment by ltrzesniewski** [comment]
Thanks! 👍

**Comment by ltrzesniewski** [comment]
Hi again @kirillgla, I noticed an issue in 2020.2 EAP 2 and I suppose it's related to this.

A call to `TransformText` yields an "ambiguous invocation" error:

![image](https://user-images.githubusercontent.com/7913492/84751786-40db1200-afbd-11ea-8158-8b76c48f0e13.png)

![image](https://user-images.githubusercontent.com/7913492/84751839-52bcb500-afbd-11ea-8283-4e9abfb76bac.png)

![image](https://user-images.githubusercontent.com/7913492/84751907-67994880-afbd-11ea-8197-2d7122393d94.png)

**Comment by ltrzesniewski** [comment]
I'm unable to reproduce this issue in a new solution, but here's some stuff I noticed:

 - The template code was generated by the VS T4 generator
 - When I regenerated it with the ForTea generator, the error disappeared
 - That _"Create the template output"_ documentation summary is generated in the template code by the T4 generator (both the VS and the ForTea one).
 - Rider doesn't display the _"Create the template output"_ summary in the `TransformText` call tooltip when the error does not appear.
 - A custom base class was used by the template (I don't know if that's important)

I hope that helps.

**Comment by kirillgla** [comment]
Thank you for the report! I'll investigate that
